service/k8sclient/k8client.go

```diff
@@ -525,19 +539,45 @@ func (c *K8Client) CreatePSMDBCluster(ctx context.Context, params *PSMDBParams)
 	return c.kubeCtl.Apply(ctx, res)
 }
 
+// GetPSMDBCluster Returns a PSMDB cluster by name.
+func (c *K8Client) GetPSMDBCluster(ctx context.Context, name string) (*PSMDBCluster, error) {
+	var cluster perconaServerMongoDB
+	err := c.kubeCtl.Get(ctx, string(perconaServerMongoDBKind), name, &cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	val := &PSMDBCluster{
+		Name:  cluster.Name,
+		State: psmdbStatesMap[cluster.Status.Status],
+		Size:  cluster.Spec.Replsets[0].Size,
+	}
+
+	if cluster.Spec.Replsets[0].Resources != nil {
+		val.Replicaset = &Replicaset{
+			ComputeResources: c.getComputeResources(*cluster.Spec.Replsets[0].Resources),
+		}
+	}
+
+	return val, nil
+}
+
 // UpdatePSMDBCluster changes size of provided percona server for mongodb cluster.
 func (c *K8Client) UpdatePSMDBCluster(ctx context.Context, params *PSMDBParams) error {
 	var cluster perconaServerMongoDB
 	err := c.kubeCtl.Get(ctx, string(perconaServerMongoDBKind), params.Name, &cluster)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "UpdatePSMDBCluster get error")
 	}
 
-	for i := range cluster.Spec.Replsets {
-		cluster.Spec.Replsets[i].Size = params.Size
+	cluster.Spec.Replsets[0].Name = params.Name
+	cluster.Spec.Replsets[0].Size = params.Size
+
+	if params.Replicaset != nil {
+		cluster.Spec.Replsets[0].Resources = c.setComputeResources(params.Replicaset.ComputeResources)
```

Comment:
it might override both previous values

Reply:
No. `c.setComputeResources` handles that.

Reply:
No. setComputeResources handles it .

Reply:
Please check one more time. Set value for Memory and update CPU. Memory value should not be changed after update

Reply:
let's use `c.updateComputeResources` here too.